utils.sh

```diff
@@ -380,18 +380,45 @@ EOF
 function verify_pull_secret() {
   # Do some PULL_SECRET sanity checking
   if [[ "${OPENSHIFT_RELEASE_IMAGE}" == *"registry.svc.ci.openshift.org"* ]]; then
-      if [[ "${PULL_SECRET}" != *"registry.svc.ci.openshift.org"* ]]; then
-          echo "Please get a valid pull secret for registry.svc.ci.openshift.org."
+      if [[ -z "${CI_TOKEN}"  ]]; then
+          error "Please login to https://api.ci.openshift.org and copy the token from the login command from the menu in the top right corner to set CI_TOKEN."
           exit 1
       fi
   fi
 
-  if [[ "${PULL_SECRET}" != *"cloud.openshift.com"* ]]; then
-      echo "Please get a valid pull secret for cloud.openshift.com."
+  if ! grep -q cloud.openshift.com "${PERSONAL_PULL_SECRET}"; then
+      error "Get a valid pull secret (json string) from https://cloud.redhat.com/openshift/install/pull-secret"
       exit 1
   fi
 }
 
+function write_pull_secret() {
+    if [ ${#PULL_SECRET} != 0 ]; then
+        echo "Using PULL_SECRET variable. Consider switching to PULL_SECRET and CI_TOKEN."
```

Comment:
...switching to PERSONAL_PULL_SECRET ...

Reply:
Refactoring is the worst.